ui/README.md

```diff
@@ -22,19 +22,20 @@ Here are a few of the benefits this provides:
   * duration
   * error
 
-If you want to add custom telemetry proprties, use `registerCommandWithCustomTelemetry` and add your own properties or measurements:
+If you want to add custom telemetry proprties, use the action's context and add your own properties or measurements:
 ```typescript
-actionHandler.registerCommandWithCustomTelemetry('yourExtension.Refresh', async (properties: TelemetryProperties, measurements: TelemetryMeasurements) => {
-    properties.customProp = "example prop";
-    measurements.customMeas = 49;
+actionHandler.registerCommand('yourExtension.Refresh', function (this: IActionContext): void {
+    this.properties.customProp = "example prop";
+    this.measurements.customMeas = 49;
 });
 ```
 
-Finally, you can also register events. The main difference is that your callback must take in the `trackTelemetry` parameter. Events are not tracked by default (since they can happen very frequently). You must call `trackTelemetry()` to signal that this event is indeed handled by your extension. For example, if your extension only handles `json` files in the `onDidSaveTextDocument`, it might look like this:
+Finally, you can also register events. By default, every event is tracked in telemetry. It is *highly recommended* to leverage the IActionContext.sendTelemetry parameter to filter only the events that apply to your extension. For example, if your extension only handles `json` files in the `onDidSaveTextDocument`, it might look like this:
 ```typescript
-actionHandler.registerEvent('yourExtension.onDidSaveTextDocument', vscode.workspace.onDidSaveTextDocument, (trackTelemetry: () => void, doc: vscode.TextDocument) => {
+actionHandler.registerEvent('yourExtension.onDidSaveTextDocument', vscode.workspace.onDidSaveTextDocument, async function (this: IActionContext, doc: vscode.TextDocument): Promise<void> {
```

Comment:
Why not an arrow function?

Reply:
You can't specify the type of "this" in an arrow function. For example, the [tslint rules](https://github.com/Microsoft/tslint-microsoft-contrib) explicitly call it out as an exception to their "no-function-expression" rule:
> Function expressions that contain a 'this' reference are allowed and will not create a failure.

[More docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions):
> An arrow function expression has a shorter syntax than a function expression and does not have its own this

Reply:
But why use this at all?  Isn't it better to be explicit about the parameters?

Reply:
The benefit of using 'this' IMO is that you can ignore `IActionContext` if you don't need to specify any custom properties, measurements, etc. (which I think is the more common case). I modeled it after the mocha tests where you can use ISuiteCallbackContext, etc. to do things like `this.setTimeout`

In other words, it was purely for the sake of convenience. If you're worried it'll be confusing or something (I was honestly on the fence about this) - I'm fine to make it a required parameter each time

Reply:
I don't get an error when passing in () => {} when the IActionContext argument is explicitly specified. If you're getting a lint error from that, though, then I agree being able to leave it out is better.